packages/polaris-viz/src/components/FunnelChartNext/Chart.tsx

```diff
@@ -0,0 +1,299 @@
+import type {ReactNode} from 'react';
+import {Fragment, useMemo, useState} from 'react';
+import {scaleBand, scaleLinear} from 'd3-scale';
+import type {
+  DataSeries,
+  XAxisOptions,
+  YAxisOptions,
+} from '@shopify/polaris-viz-core';
+import {
+  uniqueId,
+  LinearGradientWithStops,
+  useChartContext,
+} from '@shopify/polaris-viz-core';
+
+import {useFunnelBarScaling} from '../../hooks';
+import {
+  FunnelChartConnector,
+  FunnelChartConnectorGradient,
+} from '../shared/FunnelChartConnector';
+import {FunnelChartSegment} from '../shared';
+import {SingleTextLine} from '../Labels';
+import {ChartElements} from '../ChartElements';
+
+import {
+  FunnelChartXAxisLabels,
+  Tooltip,
+  FunnelTooltip,
+  TooltipWithPortal,
+} from './components';
+import type {FunnelChartNextProps} from './FunnelChartNext';
+import {
+  TOOLTIP_WIDTH,
+  LABELS_HEIGHT,
+  PERCENTAGE_SUMMARY_HEIGHT,
+  LINE_GRADIENT,
+  PERCENTAGE_COLOR,
+  LINE_OFFSET,
+  LINE_WIDTH,
+  GAP,
+  SHORT_TOOLTIP_HEIGHT,
+  TOOLTIP_HEIGHT,
+  SEGMENT_WIDTH_RATIO,
+  TOOLTIP_HORIZONTAL_OFFSET,
+} from './constants';
+
+export interface ChartProps {
+  data: DataSeries[];
+  tooltipLabels: FunnelChartNextProps['tooltipLabels'];
+  xAxisOptions: Required<XAxisOptions>;
+  yAxisOptions: Required<YAxisOptions>;
+  enableScaling: boolean;
+  renderScaleIconTooltipContent?: () => ReactNode;
+  percentageFormatter?: (value: number) => string;
```

Comment:
### Update

We aligned on a new interface as we didn't want to complicate it with x/y axis props when this chart doesn't even have them. 

```ts
export interface ChartProps {
  data: DataSeries[];
  seriesNameFormatter: LabelFormatter; // this would be Sessions, Sessions with cart addition, etc
  labelFormatter: LabelFormatter; // this would be 454,662 etc
  percentageFormatter: LabelFormatter; // applies to all percentages 
  tooltipLabels: {
    reached: string;
    dropped: string;
  };
}
```
- `labelFormatter` is an existing type, used in DonutChart, NormalizedChart, etc. It is used for strings and numbers.
- `seriesNameFormatter` and `labelFormatter` already exist at the top level of those two components, so the only new concept we are adding is the percentage one, and it follows the existing patterns
- there is no need to complicate the interface with xAxis and yAxis when our charts don't even have them